docs/content/guides/dagster/local_to_production_tutorial.mdx

```diff
@@ -0,0 +1,395 @@
+---
+title: Development to Production Workflow | Dagster
+description: This guide walks through how to transition your code from local development to production
+---
+
+# Development to Production Workflow
+
+In this example, we will walk through how to transition your assets and jobs from local development, to unit testing, and production deployments. We will cover swapping out resources and IO managers and setting run config based on the environment where your code will be executed.
+
+If you’re new to dagster, here’s an overview of the main concepts we’ll be using in this example (if you’re already familiar with dagster, feel free to skip to the next section):
+
+- asset - an asset is a software object that models a data asset. The prototypical example is a table in a database or a file in cloud storage.
+- resource - a resource is an object that models a connection to a (typically) external service. Resources can be shared between assets and different implementations of resources can be swapped in depending on environment. For example, a resource may provide methods to send messages in Slack.
+- IO manager - an IO manager is a special kind of resource that handles storing and loading assets. For example, if we wanted to store assets in S3, we would use dagster’s built in S3 IO manager.
+- run config - assets and resources sometimes require configuration to set certain values (like the password to a database). Run config allows you to set these values at run time. In this example, we will also use an API to set some default run configuration.
+
+## Problem Statement
+
+Let’s say we’ve been tasked with fetching data from Hacker News and splitting it into two datasets: one containing all of the data about stories and one containing all of the data about comments. We have the following additional requirements:
+
+- For local development, we want to store the datasets as tables in a development snowflake database called `SANDBOX` that has a schema for each user.
+- For production, we want to store the datasets as tables in a production snowflake database called `PRODUCTION` that has a schema called `HACKER_NEWS`.
+- For testing, we don’t want to fetch data from Hacker News, instead we want to provide seed data so we can assert that the data was split into the stories and comments datasets correctly.
```

Comment:
I feel we could make this section above the "If you're new to dagster...", and maybe center it around the "why's" such as the current pain points that you can't simply run the same code across different environments without the fear of breaking something that you are not aware of. 

i think the sda guide ([why use sda](https://dagster-btbtnhnr6-elementl.vercel.app/guides/dagster/enriching-with-software-defined-assets#why-use-software-defined-assets)) did a good job attracting audience who has lived in the same pain. we don't have to expand too much like the sda guide did but several painful scenarios might help the reader understand why a good workflow matters to them.

cc @erinkcochran87 may have thoughts on how we structuring the info too.

Reply:
I agree with bumping this section up and including pain points. Perhaps the general structure could be something like this:

- Pain points, such as not being able to run the same code across environments
- Dagster concepts used in the guide and how they help address pain points
- Brief summary of the example we'll use to demonstrate the pain points and how to solve them

Reply:
i reorganized this section based on your ideas, let me know what you think!